pkg/kubelet/status/status_manager.go

```diff
@@ -183,23 +189,25 @@ func isPodStatusByKubeletEqual(oldStatus, status *v1.PodStatus) bool {
 }
 
 func (m *manager) Start() {
-	// Don't start the status manager if we don't have a client. This will happen
-	// on the master, where the kubelet is responsible for bootstrapping the pods
-	// of the master components.
-	if m.kubeClient == nil {
-		klog.InfoS("Kubernetes client is nil, not starting status manager")
-		return
-	}
-
+	// Create pod allocation checkpoint manager even if we don't have a client to allow local get/set of ResourcesAllocated & Resize
 	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
 		stateImpl, err := state.NewStateCheckpoint(m.stateFileDirectory, podStatusManagerStateFile)
 		if err != nil {
+			//TODO(vinaykul,InPlacePodVerticalScaling): Check if this error should be a critical (stop kubelet) failure
```

Comment:
we're going to absolutely flood the kubelet error log when the feature is enabled if state ends up nil

Reply:
This particular log line? Start() only gets called once when [kubelet is spun up](https://github.com/kubernetes/kubernetes/blob/f461527491effcdbe23814b7481f46723cddda9e/pkg/kubelet/kubelet.go#L1546L1547). Or did I miss something?

I do somehow feel that checkpoint state allocation failure is worthy of panic and get rid of the Errorf in SetPodAllocation and SetPodResizeStatus (which will flood the logs), but I'll go with SME opinions about adding panic here.

Reply:
no, the places we return errors if state is nil

Reply:
Got it. The more I think about it, the more I feel that hitting the panic button here and getting rid of nil checks is the better way to go. Let me try it out and see how it looks in code.